frontend/src/scenes/session-recordings/player/sessionRecordingPlayerLogic.ts

```diff
@@ -487,10 +503,54 @@ export const sessionRecordingPlayerLogic = kea<sessionRecordingPlayerLogicType>(
     windowValues({
         isSmallScreen: (window: any) => window.innerWidth < getBreakpoint('md'),
     }),
-    events(({ values, actions }) => ({
+    events(({ values, actions, cache }) => ({
         beforeUnmount: () => {
             values.player?.replayer?.pause()
             actions.setPlayer(null)
+            if (cache.originalWarning) {
+                console.warn = cache.originalWarning
+            }
+            if (cache.errorHandler) {
+                window.removeEventListener('error', cache.errorHandler)
+            }
+            eventUsageLogic.actions.reportRecordingViewedSummary({
```

Comment:
Does it make sense to have the same reporting happening wherever we mount the sessionRecordingPlayer logic? I think whenever sessionRecordingPlayer is unmounted, the corresponding sessionRecordingDataLogic should be unmounted as well, which would make this listener redundant. 

If we remove this listener, we can also clean up boilerplate by removing `loadMetaTimeMs`, `loadFirstSnapshotTimeMs` and `loadAllSnapshotTimeMs` and calculating those just in time on sessionRecordingData unmount.

Reply:
Alex and I touched base on this one. We still want the analytics from `sessionRecordingDataLogic` to fire for now until we move graphs to the new stuff. So having some similar logic in both places makes sense for now.